mempool README.txt

```diff
@@ -0,0 +1,163 @@
+Raylib Memory Pool
+By Kevin 'Assyrianic' Yonan @ https://github.com/assyrianic
+
+About:
+	The Raylib Memory Pool is a quick, efficient, and minimal free list & stack-based allocator.
+
+Purpose:
+	Raylib Memory Pool's purpose is the following list:
+		* A quicker, efficient memory allocator alternative to `malloc` and friends.
+		* Reduce the possibilities of memory leaks for beginner developers using Raylib.
+		* Being able to flexibly range check memory if necessary.
+
+Data Implementation:
+	the memory pool encapsulates two public structs:
+	`freeList` which is an abstracted doubly linked list consisting of a `head` and `tail` pointer to `struct MemNode`, a `len` that tracks the amount of nodes the linked list holds, `maxNodes` which is used for auto-defragging (explained below), and `autoDefrag` which controls whether auto-defragging will execute or not.
+	```c
+	typedef struct MemNode {
+		size_t size;
+		struct MemNode *next, *prev;
+	} MemNode;
+
+	typedef struct AllocList {
+		struct MemNode *head, *tail;
+		size_t len, maxNodes;
+		bool autoDefrag : 1;
+	} AllocList;
+	```
+	
+	`Stack` which is an array-based stack consisting of a byte pointer to the entire array `mem`, the base pointer `base` which changes position when allocating, deallocating, or defragging, and `size` which tracks how large the entire stack buffer is.
+	```c
+	typedef struct Stack {
+		uint8_t *mem, *base;
+		size_t size;
+	} Stack;
+	```
+	
+	```c
+	typedef struct MemPool {
+		struct AllocList freeList;
+		struct Stack stack;
+	} MemPool;
+	```
+
+Explanation & Usage:
+	The memory pool is designed to be used as a direct object.
+	We have two constructor functions:
+	```c
+	struct MemPool MemPool_Create(size_t bytes);
+	struct MemPool MemPool_FromBuffer(size_t bytes, void *buf);
+	```
+	
+	To which you create a `struct MemPool` instance and give the function a max amount of memory you wish or require for your data.
+	Remember not to exceed that memory amount or the allocation functions of the allocator will give you a NULL pointer.
+	
+	
+	So we create a pool that will malloc 10K bytes.
+	```c
+	struct MemPool pool = MemPool_Create(10000);
+	```
+	
+	When we finish using the pool's memory, we clean it up by using `MemPool_Destroy`.
+	```c
+	MemPool_Destroy(&pool);
+	```
+	
+	Alternatively, if you're not in a position to use any kind of dynamic allocation from the operating system, you have the option to utilize an existing buffer as memory for the mempool:
+	```c
+	char mem[64000];
+	struct MemPool pool = MemPool_FromBuffer(sizeof mem, &mem[0]);
```

Comment:
`&mem[0]` => `mem`. The first variant is unnecessarily verbose.

Reply:
I did that to be intentionally verbose, I'll remove it since you asked.